tensorlayer/layers/core.py

```diff
@@ -60,73 +59,14 @@ def flatten_reshape(variable, name='flatten'):
     return tf.reshape(variable, shape=[-1, dim], name=name)
 
 
+@deprecated("2018-06-30", "TensorLayer relies on TensorFlow to check naming.")
 def clear_layers_name():
-    """Clear all layer names in LayersConfig
-
-    Examples
-    ---------
-    Clean the current graph and try to re-define model.
-
-    >>> for .... (different model settings):
-    >>>    with tf.Graph().as_default() as graph:   # clear all variables of TF
-    >>>       tl.layers.clear_layers_name()         # clear all layer name of TL
-    >>>       sess = tf.InteractiveSession()
-    >>>       # define and train a model here
-    >>>       sess.close()
-
-    Enable reusing layer names.
-
-    >>> net = tl.layers.InputLayer(x, name='input_layer')
-    >>> net = tl.layers.DenseLayer(net, n_units=800, name='relu1')
-    ...
-    >>> tl.layers.clear_layers_name()
-    >>> net2 = tl.layers.InputLayer(x, name='input_layer')
-    >>> net2 = tl.layers.DenseLayer(net2, n_units=800, name='relu1')
-
-    """
-    LayersConfig._layer_name_list = []
+    logging.warning('this method is DEPRECATED and has no effect, please remove it from your code.')
 
 
+@deprecated("2018-06-30", "TensorLayer relies on TensorFlow to check name reusing.")
 def set_name_reuse(enable=True):
-    """Enable or disable reuse layer name.
-
-    By default, each layer must has unique
-    name. When you want two or more input placeholder (inference) share the same
-    model parameters, you need to enable layer name reuse, then allow the
-    parameters have same name scope.
-
-    Parameters
-    ----------
-    enable : boolean
-        Enable or disable name/layer reuse, None means False.
-
-    Examples
-    --------
-    >>> def embed_seq(input_seqs, is_train, reuse):
-    >>>    with tf.variable_scope("model", reuse=reuse):
-    >>>         tl.layers.set_name_reuse(reuse)
-    >>>         net = tl.layers.EmbeddingInputlayer(
-    ...                     inputs = input_seqs,
-    ...                     vocabulary_size = vocab_size,
-    ...                     embedding_size = embedding_size,
-    ...                     name = 'e_embedding')
-    >>>         net = tl.layers.DynamicRNNLayer(net,
-    ...                     cell_fn = tf.contrib.rnn.BasicLSTMCell,
-    ...                     n_hidden = embedding_size,
-    ...                     dropout = (0.7 if is_train else None),
-    ...                     initializer = w_init,
-    ...                     sequence_length = tl.layers.retrieve_seq_length_op2(input_seqs),
-    ...                     return_last = True,
-    ...                     name = 'e_dynamicrnn')
-    >>>    return net
-    >>>
-    >>> net_train = embed_seq(t_caption, is_train=True, reuse=False)
-    >>> net_test = embed_seq(t_caption, is_train=False, reuse=True)
-
-    - see ``tutorial_ptb_lstm.py`` for example.
-
-    """
-    LayersConfig._name_reuse = enable
+    logging.warning('this method is DEPRECATED and has no effect, please remove it from your code.')
```

Comment:
Actually Hao was right, use `pass` should be enough, because TensorFlow would log the warning.
And I think this code will crash because we don't have a `warning` method on `logging` yet.

Reply:
I see the `warning` method was added later, then it's fine here.